File: app/src/main/java/com/leonov_dev/currencyconverter/CategoryAdapter.java
package com.leonov_dev.currencyconverter;

import android.content.Context;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class CategoryAdapter extends FragmentPagerAdapter {

    private Context mContext;

    public CategoryAdapter(Context context, FragmentManager fm) {
        super(fm);
        mContext = context;

    }

    @Override
    public Fragment getItem(int position) {
        if (position == 0){
            return new StockFragment();
        }else {
            return new ConverterFragment();
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        if (position == 0){
            return mContext.getString(R.string.StockTitle);
        }else {
            return mContext.getString(R.string.ConverterTitle);
        }
    }

    @Override
    public int getCount() {
        return 2;
    }
}

File: app/src/main/java/com/leonov_dev/currencyconverter/ui/home/stock/CategoryAdapter.kt
package com.leonov_dev.currencyconverter.ui.home.stock

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.leonov_dev.currencyconverter.R
import com.leonov_dev.currencyconverter.ui.home.converter.ConverterFragment

class CategoryAdapter(private val mContext: Context, fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return if (position == 0) {
            StockFragment()
        } else {
            ConverterFragment()
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return if (position == 0) {
            mContext.getString(R.string.StockTitle)
        } else {
            mContext.getString(R.string.ConverterTitle)
        }
    }

    override fun getCount(): Int = 2
}
